File: Java (client)/ClientServerWithJava/Persistence/src/main/java/ro/mpp2024/TicketDBRepository.java
package ro.mpp2024;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class TicketDBRepository implements ITicketRepository {

    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger(TicketDBRepository.class);

    public TicketDBRepository(Properties props){
        logger.info("Initializing TicketDBRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public Ticket save(Ticket entity) {
        logger.traceEntry("saving ticket {} ",entity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("insert into tickets(matchID, price, clientName, numberOfSeats) values (?, ?, ?, ?) ")){
            statement.setInt(1, entity.getMatchID());
            statement.setInt(2, entity.getPrice());
            statement.setString(3, entity.getClientName());
            statement.setInt(4, entity.getNumberOfSeats());
            int result = statement.executeUpdate();
            return entity;
        }catch(SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return null;
    }

    @Override
    public Ticket delete(Integer aLong) {
        return null;
    }

    @Override
    public Ticket update(Ticket entity) {
        return null;
    }

    @Override
    public Ticket findOne(Integer aLong) {
        logger.traceEntry("finding ticket with id {} ",aLong);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("select * from tickets where id = ?")) {
            statement.setInt(1, aLong);
            try (var result = statement.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String clientName = result.getString("clientName");
                    int matchID = result.getInt("matchID");
                    int price = result.getInt("price");
                    int numberOfSeats = result.getInt("numberOfSeats");
                    Ticket ticket = new Ticket(id, matchID, price, clientName, numberOfSeats);
                    return ticket;
                }
            }
        }catch (SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return null;
    }

    @Override
    public Iterable<Ticket> findAll() {
        logger.traceEntry("finding all tickets");
        List<Ticket> tickets = new ArrayList<>();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("select * from tickets")){
            try(var result = statement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("id");
                    String clientName = result.getString("clientName");
                    int matchID = result.getInt("matchID");
                    int price = result.getInt("price");
                    int numberOfSeats = result.getInt("numberOfSeats");
                    Ticket ticket = new Ticket(id, matchID, price, clientName, numberOfSeats);
                    tickets.add(ticket);
                }
            }}
        catch (SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return tickets;
    }

}

File: C# (server)/Persistence/TicketDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using Model;
using log4net;
using log4net.Util;

namespace Persistence
{
    public class TicketDBRepository : ITicketRepository
    {
        private static readonly ILog logger = LogManager.GetLogger("TicketDBRepository");

        IDictionary<String, string> props;

        public TicketDBRepository(IDictionary<String, string> props)
        {
            logger.Info("Creating TicketDBRepository");
            this.props = props;
        }

        public IEnumerable<Ticket> findAll()
        {
            logger.Info("Finding all tickets");
            IList<Ticket> tickets = new List<Ticket>();
            IDbConnection connection = DBUtils.getConnection(props);
            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id,matchID, price, clientName, numberOfSeats from tickets";
                using (var result = command.ExecuteReader())
                {
                    while (result.Read())
                    {
                        int id = result.GetInt32(0);
                        int matchID = result.GetInt32(1);
                        int price = result.GetInt32(2);
                        string clientName = result.GetString(3);
                        int numberOfSeats = result.GetInt32(4);
                        Ticket ticket = new Ticket(id, matchID, price, clientName, numberOfSeats);
                        tickets.Add(ticket);
                        
                    }
                }
            }

            logger.InfoExt(tickets);
            return tickets;
        }

        public Ticket findOne(int id)
        {
            logger.Info("Finding ticket with id " + id);
            IDbConnection connection = DBUtils.getConnection(props);
            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id, matchID, price, clientName, numberOfSeats from tickets where id=@id";
                IDbDataParameter paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);
                using (var result = command.ExecuteReader())
                {
                    if (result.Read())
                    {
                        int idv = result.GetInt32(0);
                        int matchID = result.GetInt32(1);
                        int price = result.GetInt32(2);
                        string clientName = result.GetString(3);
                        int numberOfSeats = result.GetInt32(4);
                        Ticket ticket = new Ticket(idv, matchID, price, clientName, numberOfSeats);
                        logger.InfoExt(ticket);
                        return ticket;
                        
                    }
                }
            }

            return null;
        }

        public Ticket save(Ticket entity)
        {
            logger.Info("Saving ticket with id " + entity.id);
            IDbConnection connection = DBUtils.getConnection(props);
            try
            {
                
                using (var command = connection.CreateCommand())
                {
                    command.CommandText = "INSERT INTO tickets(matchID, price, clientName, numberOfSeats) VALUES (@match, @price, @client, @seats)";
                    IDbDataParameter paramSeats = command.CreateParameter();
                    paramSeats.ParameterName = "@seats";
                    paramSeats.Value = entity.numberOfSeats;
                    command.Parameters.Add(paramSeats);
                    IDbDataParameter paramid = command.CreateParameter();
                    paramid.ParameterName = "@match";
                    paramid.Value = entity.matchID;
                    command.Parameters.Add(paramid);
                    IDbDataParameter paramclient = command.CreateParameter();
                    paramclient.ParameterName = "@client";
                    paramclient.Value = entity.clientName;
                    command.Parameters.Add(paramclient);
                    IDbDataParameter paramprice = command.CreateParameter();
                    paramprice.ParameterName = "@price";
                    paramprice.Value = entity.price;
                    command.Parameters.Add(paramprice);
                    command.ExecuteNonQuery();
                    return entity;
                }
            }
            catch (Exception e)
            {
                logger.Error(e);
                return null;
            }

        }

        public Ticket delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public Ticket update(Ticket entity)
        {
            throw new System.NotImplementedException();
        }
    }
}